File: Vista/ConexionDialog.py
from PyQt6.QtWidgets import QDialog, QVBoxLayout, QHBoxLayout, QLabel, QLineEdit, QComboBox, QPushButton, QMessageBox, QWidget
from PyQt6.QtCore import Qt
from ConexionBD.Conexion import Conexion, TipoMotor, TipoAutenticacion

class ConexionDialog(QDialog):
    def __init__(self, parent=None):
        super().__init__(parent)
        self.setWindowTitle("Configuracion de Conexion")
        self.setModal(True)
        self.confirmado = False
        self.construirUI()
        self.setFixedSize(400, 450)

    def construirUI(self):
        layout = QVBoxLayout(self)
        layout.setContentsMargins(24, 20, 24, 20)
        layout.setSpacing(10)

        titulo = QLabel("Selecciona el motor de base de datos")
        font = titulo.font()
        font.setBold(True)
        font.setPointSize(14)
        titulo.setFont(font)
        layout.addWidget(titulo)

        def add_row(label_text, widget):
            row = QHBoxLayout()
            lbl = QLabel(label_text)
            lbl.setFixedWidth(100)
            row.addWidget(lbl)
            row.addWidget(widget)
            layout.addLayout(row)
            return lbl

        self.comboMotor = QComboBox()
        self.comboMotor.addItems(["MySQL"])
        self.comboMotor.setEnabled(False)
        add_row("Motor:", self.comboMotor)

        self.campoHost = QLineEdit("localhost")
        add_row("Host:", self.campoHost)

        self.campoPuerto = QLineEdit("3306")
        self.labelPuerto = add_row("Puerto:", self.campoPuerto)

        self.campoInstancia = QLineEdit("1433")
        self.labelInstancia = add_row("Puerto TCP:", self.campoInstancia)

        self.campoBase = QLineEdit("corporativo_pos")
        add_row("Base de datos:", self.campoBase)

        self.campoUsuario = QLineEdit("root")
        self.labelUsuario = add_row("Usuario:", self.campoUsuario)

        self.campoContrasena = QLineEdit("2306")
        self.campoContrasena.setEchoMode(QLineEdit.EchoMode.Password)
        self.labelContrasena = add_row("Contrasena:", self.campoContrasena)

        self.comboMotor.currentIndexChanged.connect(self.actualizarVista)

        btnLayout = QHBoxLayout()
        btnLayout.addStretch()
        
        btnSalir = QPushButton("Salir")
        btnSalir.clicked.connect(self.reject)
        
        btnConectar = QPushButton("Conectar")
        btnConectar.clicked.connect(self.intentarConexion)
        
        btnLayout.addWidget(btnSalir)
        btnLayout.addWidget(btnConectar)
        
        layout.addLayout(btnLayout)
        self.actualizarVista()

    def actualizarVista(self):
        self.labelInstancia.setVisible(False)
        self.campoInstancia.setVisible(False)
        self.labelPuerto.setVisible(True)
        self.campoPuerto.setVisible(True)
        self.labelUsuario.setVisible(True)
        self.campoUsuario.setVisible(True)
        self.labelContrasena.setVisible(True)
        self.campoContrasena.setVisible(True)
        self.campoPuerto.setText("3306")
        self.campoUsuario.setText("root")

    def intentarConexion(self):
        host = self.campoHost.text().strip()
        base = self.campoBase.text().strip()
        puerto = self.campoPuerto.text().strip()
        usr = self.campoUsuario.text().strip()
        pass_ = self.campoContrasena.text()

        if not host or not base or not puerto or not usr:
            QMessageBox.warning(self, "Aviso", "Completa todos los campos.")
            return

        Conexion.configurar(TipoMotor.MYSQL, TipoAutenticacion.CREDENCIALES, host, puerto, base, usr, pass_)

        if Conexion.getConexion() is not None:
            self.confirmado = True
            self.accept()
        else:
            QMessageBox.critical(self, "Error de Conexion", "No se pudo conectar.\nRevisa los datos e intenta de nuevo.")
            Conexion.cerrar()

    def fueConfirmado(self):
        return self.confirmado

File: Vista/ConexionDialog.cs
using Microsoft.Maui.Controls;
using Microsoft.Maui.Graphics;
using System;
using System.Threading.Tasks;
using punto_de_venta_C_.ConexionBD;

namespace punto_de_venta_C_.Vista
{
    public class ConexionDialog : ContentPage
    {
        private bool confirmado = false;
        private Entry campoHost;
        private Entry campoPuerto;
        private Entry campoBase;
        private Entry campoUsuario;
        private Entry campoContrasena;
        private TaskCompletionSource<bool> tcs = new TaskCompletionSource<bool>();

        public ConexionDialog()
        {
            Title = "Configuracion de Conexion";
            NavigationPage.SetHasNavigationBar(this, false);
            construirUI();
        }

        private void construirUI()
        {
            var layout = new VerticalStackLayout
            {
                Spacing = 12,
                Padding = new Thickness(24),
                BackgroundColor = Estilos.BG_BLANCO,
                WidthRequest = 450,
                HorizontalOptions = LayoutOptions.Center,
                VerticalOptions = LayoutOptions.Center
            };

            var titulo = new Label
            {
                Text = "Selecciona el motor de base de datos",
                FontSize = 16,
                FontAttributes = FontAttributes.Bold,
                TextColor = Estilos.TEXTO_PRINCIPAL
            };
            layout.Add(titulo);

            layout.Add(new Label { Text = "Motor (Solo MySQL):", FontSize = Estilos.FUENTE_XS_SIZE, TextColor = Estilos.TEXTO_TENUE });
            var entryMotor = new Entry { Text = "MySQL", IsReadOnly = true, BackgroundColor = Estilos.BG_ZINC_100 };
            layout.Add(entryMotor);

            layout.Add(new Label { Text = "Host:", FontSize = Estilos.FUENTE_XS_SIZE, TextColor = Estilos.TEXTO_TENUE });
            campoHost = new Entry { Text = "localhost" };
            layout.Add(campoHost);

            layout.Add(new Label { Text = "Puerto:", FontSize = Estilos.FUENTE_XS_SIZE, TextColor = Estilos.TEXTO_TENUE });
            campoPuerto = new Entry { Text = "3306", Keyboard = Keyboard.Numeric };
            layout.Add(campoPuerto);

            layout.Add(new Label { Text = "Base de datos:", FontSize = Estilos.FUENTE_XS_SIZE, TextColor = Estilos.TEXTO_TENUE });
            campoBase = new Entry { Text = "corporativo_pos" };
            layout.Add(campoBase);

            layout.Add(new Label { Text = "Usuario:", FontSize = Estilos.FUENTE_XS_SIZE, TextColor = Estilos.TEXTO_TENUE });
            campoUsuario = new Entry { Text = "root" };
            layout.Add(campoUsuario);

            layout.Add(new Label { Text = "Contrasena:", FontSize = Estilos.FUENTE_XS_SIZE, TextColor = Estilos.TEXTO_TENUE });
            campoContrasena = new Entry { Text = "2306", IsPassword = true };
            layout.Add(campoContrasena);

            var panelBotones = new Grid
            {
                ColumnDefinitions = new ColumnDefinitionCollection
                {
                    new ColumnDefinition { Width = GridLength.Star },
                    new ColumnDefinition { Width = 100 },
                    new ColumnDefinition { Width = 8 },
                    new ColumnDefinition { Width = 100 }
                }
            };

            var btnSalir = Estilos.botonPeligro("Salir");
            btnSalir.Clicked += (s, e) => Environment.Exit(0);
            panelBotones.Add(btnSalir, 1, 0);

            var btnConectar = Estilos.botonPrimario("Conectar");
            btnConectar.Clicked += (s, e) => intentarConexion();
            panelBotones.Add(btnConectar, 3, 0);

            layout.Add(panelBotones);

            Content = new Grid
            {
                BackgroundColor = Estilos.BG_CLARO,
                Children = { layout },
                HorizontalOptions = LayoutOptions.Center,
                VerticalOptions = LayoutOptions.Center
            };
        }

        private async void intentarConexion()
        {
            string host = campoHost.Text?.Trim() ?? "";
            string baseDatos = campoBase.Text?.Trim() ?? "";
            string puerto = campoPuerto.Text?.Trim() ?? "";
            string usr = campoUsuario.Text?.Trim() ?? "";
            string pass = campoContrasena.Text ?? "";

            if (string.IsNullOrEmpty(host) || string.IsNullOrEmpty(baseDatos) || string.IsNullOrEmpty(puerto) || string.IsNullOrEmpty(usr))
            {
                await DisplayAlert("Aviso", "Completa todos los campos.", "OK");
                return;
            }

            Conexion.configurar(Conexion.TipoMotor.MYSQL, Conexion.TipoAutenticacion.CREDENCIALES, host, puerto, baseDatos, usr, pass);

            var conn = Conexion.getConexion();
            if (conn != null)
            {
                confirmado = true;
                tcs.SetResult(true);
            }
            else
            {
                await DisplayAlert("Error de Conexion", "No se pudo conectar.\nRevisa los datos e intenta de nuevo.", "OK");
                Conexion.cerrar();
            }
        }

        public async Task<bool> mostrar()
        {
            if (Application.Current != null)
            {
                Application.Current.MainPage = new NavigationPage(this);
            }
            return await tcs.Task;
        }

        public bool fueConfirmado()
        {
            return confirmado;
        }
    }
}
